pkg/kubectl/cmd/plugin.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/plugins"
+)
+
+// NewCmdForPlugin creates a command capable of running the provided plugin.
+func NewCmdForPlugin(plugin *plugins.Plugin, f cmdutil.Factory, in io.Reader, out, errout io.Writer) *cobra.Command {
+	if len(plugin.Name) == 0 ||
+		len(plugin.ShortDesc) == 0 ||
+		len(plugin.Command) == 0 {
+		return nil
+	}
+
+	runner := f.PluginRunner()
+
+	return &cobra.Command{
+		Use:     plugin.Name,
```

Comment:
Not all of this will be done in one PR.  We're following a milestone doc.
Resolving the details of dealing w/ native command collisions is in the next milestone and is frankly a relatively simple thing.  Also coming is making a tunnel available.